clients/src/test/java/org/apache/kafka/common/requests/RequestTestUtils.java

```diff
@@ -208,10 +233,10 @@ public static MetadataResponse metadataUpdateWith(final String clusterId,
             for (int i = 0; i < numPartitions; i++) {
                 TopicPartition tp = new TopicPartition(topic, i);
                 Node leader = nodes.get(i % nodes.size());
-                List<Integer> replicaIds = Collections.singletonList(leader.id());
+                List<Integer> replicaIds = leaderOnly ? Collections.singletonList(leader.id()) : nodes.stream().map(Node::id).collect(Collectors.toList());
```

Comment:
Updated to use an explicit flag to use all replicas vs just the leader.  @hachikuji